editor/src/clj/dynamo/graph.clj

```diff
@@ -160,8 +160,9 @@
          basis (is/basis system)
          id-generators (is/id-generators system)
          override-id-generator (is/override-id-generator system)
+         tx-data-context-map (or (:tx-data-context-map opts) {})
          metrics-collector (:metrics opts)
-         transaction-context (it/new-transaction-context basis id-generators override-id-generator metrics-collector)
+         transaction-context (it/new-transaction-context basis id-generators override-id-generator tx-data-context-map metrics-collector)
```

Comment:
The `:tx-data-context` field inside the `transaction-context` hosts an `atom` that can be used to communicate state changes inside the transaction. Previously we were using it to keep track of the node-ids of `ResourceNodes` created as a result of a property setter referring to a resource we haven't picked up yet to make sure we do not create duplicates. For example, if several sprites would refer to a missing material resource, we want all of them to refer to the same defective resource node, not create duplicate defective resource nodes.

This `:tx-data-context` atom is accessible from the `evaluation-context` supplied to all property `set` functions.

You can now supply an initial state for the map in the `:tx-data-context`, and we use it to host the set of property endpoints edited by the user using the Property Editor. This enables use to check inside a property `set` function if we're being invoked as a result of the user editing a particular property with the help of the new `properties/user-edit?` function.